File: pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <stdio.h>

int main(void)
{   
    // Get input from user
    string text = get_string("Text: ");
    
    // To record the number of letters, words and sentences.
    float letter = 0.0;
    float sentence = 0.0;
    float words = 1.0;
    
    for (int i = 0; i < strlen(text); i++)
    {
        // To find the number of alphabetical letters in the text. 
        if isalpha(text[i])
        {
            letter++;
        }
        // To find the number of sentences.
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            sentence++;
        }
        // To find the number of words.
        if isblank(text[i])
        {
            words++;
        }
    }
    
    // To calculate both the average letters and sentences per 100 words.
    float L = (letter / words) * 100.0;
    float S = (sentence / words) * 100.0;
   
    // Coleman liau index formula
    int index = round(0.0588 * L - 0.296 * S - 15.8);
   
    // To determine whether if grade is above 16, below 1 or in between.
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: pset6/readabilitypy/readability.py
from cs50 import get_string

# Letter, sentence and word counter.
letter = 0
sentence = 0
words = 1

text = get_string("Text: ")

for i in text:
    # Checks for number of letters
    if i.isalpha():
        letter += 1
    # Checks for spaces which means a word has been passed by.
    if i == " ":
        words += 1
    # Checks for end of sentence.
    if i == "." or i == "!" or i == "?":
        sentence += 1

# Coleman liau index formula
L = (letter / words) * 100
S = (sentence / words) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# Checks if grade is 16+, below 1 or in between.
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")